python/pyarrow/types.pxi

```diff
@@ -5333,7 +5334,7 @@ def bool8():
     pyarrow.Table
     unknown_col: extension<arrow.bool8>
     ----
-    unknown_col: [[True, False, True, True, null]]
+    unknown_col: [[-1,0,1,2,null]]
```

Comment:
Sidenote: this is a good illustration for that we should ideally have a way to let the extension type control this string representation

Reply:
This is a great point and certainly something I would have liked to have when going through this implementation. I'll open an issue for it.

Reply:
We already have https://github.com/apache/arrow/issues/36648 covering that I think